src/kalshi.rs

```diff
@@ -1,214 +1,231 @@
             threshold: requirements.min_days_to_resolution,
         });
     }
-    if question.time_to_resolution() > Duration::days(requirements.max_days_to_resolution) {
+    if market.time_to_resolution() > Duration::days(requirements.max_days_to_resolution) {
         return Err(KalshiCheckFailure::ResolvesTooLate {
-            days_remaining: question.time_to_resolution().num_days(),
+            days_remaining: market.time_to_resolution().num_days(),
             threshold: requirements.max_days_to_resolution,
         });
     }
-    if question.age() > Duration::days(requirements.max_age_days) {
+    if market.age() > Duration::days(requirements.max_age_days) {
         return Err(KalshiCheckFailure::TooOld {
-            age_days: question.age().num_days(),
+            age_days: market.age().num_days(),
             threshold: requirements.max_age_days,
         });
     }
-    if (100 - question.get_market().yes_ask) > requirements.max_confidence * 100
-    || question.get_market().yes_bid > requirements.max_confidence * 100 {
+    if (100 - market.yes_ask) as f64 > requirements.max_confidence * 100.0
+    || market.yes_bid as f64 > requirements.max_confidence * 100.0 {
         return Err(KalshiCheckFailure::TooExtreme {
-            yes_ask: question.get_market().yes_ask,
-            yes_bid: question.get_market().yes_bid,
+            yes_ask: market.yes_ask,
+            yes_bid: market.yes_bid,
             threshold: requirements.max_confidence,
         });
     }
-    if requirements.exclude_ids.contains(&question.id) {
+    if requirements.exclude_ids.contains(&market.id()) {
         return Err(KalshiCheckFailure::Banned);
     }
 
     Ok(())
 }
 
-impl KalshiQuestion {
-    pub fn age(&self) -> Duration {
-        Utc::now() - self.get_market().open_date
+/// helper function for parsing both success and error responses
+fn parse_response<T: DeserializeOwned>(resp: Response) -> Result<T, KalshiError> {
+    if resp.status().is_success() {
+        let body = resp.text().map_err(|_| KalshiError::UnexpectedResponseType)?;
+        match serde_json::from_str(&body) {
+            Ok(r) => Ok(r),
+            Err(e) => {
+                print!("Response: {}", body);
+                println!("Error parsing response from Kalshi: {}", e);
+                Err(KalshiError::UnexpectedResponseType)
+            }
+        }
+    } else {
+        let status = resp.status();
+        let error_resp: KalshiErrorResponse = resp
+            .json()
+            .map_err(|_| KalshiError::UnexpectedErrorType(status))?;
+        Err(KalshiError::ErrorResponse(status, error_resp))
     }
+}
 
-    pub fn has_matching_market(&self) -> bool {
-        self.event
-            .markets
-            .iter()
-            .find(|market| market.ticker_name == self.id)
-            .is_some()
+impl KalshiMarket {
+    pub fn id(&self) -> String {
+        self.ticker_name.clone()
```

Comment:
I think I'll leave it non-capsed, since `ticker_name` is taken untouched straight from the JSON as how the market identifies itself. I'll adjust some comments and variable names to make this clearer, and return `&str`.